lib/vanagon/component/source.rb

```diff
@@ -1,78 +1,114 @@
+require 'fustigit'
 require 'vanagon/component/source/http'
 require 'vanagon/component/source/git'
 require 'vanagon/component/source/local'
 
 class Vanagon
   class Component
     class Source
-      SUPPORTED_PROTOCOLS = ['file', 'http', 'git']
-      @@rewrite_rule = {}
+      SUPPORTED_PROTOCOLS = %w(file http https git).freeze
+      @rewrite_rules = {}
 
-      def self.register_rewrite_rule(protocol, rule)
-        if rule.is_a?(String) or rule.is_a?(Proc)
-          if SUPPORTED_PROTOCOLS.include?(protocol)
-            @@rewrite_rule[protocol] = rule
+      class << self
+        attr_reader :rewrite_rules
+
+        # Deprecate this
```

Comment:
I'm going to add deprecation notices to all of the rewrite methods in the next big PR, that will add the concept of mirror URLs. That will give me someplace else to point users for their needs.